zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGenerator.scala

```diff
@@ -334,6 +354,13 @@ class SQLPPGenerator extends AsterixQueryGenerator {
       case _ => aggFuncExpr(typeImpl.getAggregateStr(aggr.func))
     }
   }
+
+  protected def append(queryBuilder: StringBuilder, queryStr: String): Unit = {
+    if (!queryStr.isEmpty) {
```

Comment:
then let's make a good name like `appendIfQueryIsNotEmpty` or something else that describe the behavior?